conda_lock/src_parser/pyproject_toml.py

```diff
@@ -213,14 +215,14 @@ def specification_with_dependencies(
 
     if force_pypi:
         for dep in dependencies:
-            if dep.name in force_pypi:
-                dep.manager = "pip"
+            if dep.dep.name in force_pypi:
+                dep.dep.manager = "pip"
```

Comment:
Having `dep.dep` and `.to_source()` seems a bit awkward. I wonder if there's a more natural approach to SourceDependency?

Reply:
For the `dep.dep` case, we can use properties to expose common attributes of `Dependency` objects like `name`, `manager`, etc.

Reply:
Would it make sense to have `SourceDependency` as a subclass of `Dependency`?

Reply:
We'd need a `SourceURLDependency` and a `SourceVersionedDependency` in that case, right?